app/components/view_component/form/field_component.rb

```diff
@@ -60,6 +60,26 @@ def object_method_names
         end
       end
 
+      def optional?
+        return nil if object.nil?
+
+        !required?
+      end
+
+      def required?
+        return nil if object.nil?
+
+        validators.any? do |v|
+          v.is_a? ActiveRecord::Validations::PresenceValidator
+        end
+      end
+
+      def validators
```

Comment:
Idea: add support for context

```suggestion
      def validators(context: nil)
```